File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str = new StringBuilder();
        
        for(int i = 0; i < s.length(); i++){
            char ch = s.charAt(i);
            if(ch>= 48 && ch<= 57){
                str.append(ch);
            }else if(ch >= 65 && ch <= 90){
                str.append((char)(ch - 'A' + 'a'));
            }else if(ch >= 97 && ch<=122){
                str.append(ch);
            }
        }
        
        return isPalin(str.toString());
    }
    
    
    public boolean isPalin(String s){
        int i = 0;
        int j = s.length()-1;
        
        while(i<j){
            if(s.charAt(i)!=s.charAt(j)) return false;
            i++;
            j--;
        }
        return true;
    }
    
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool check(string s){
        int l = 0, r = s.size()-1;
        while(l<r){
            if(s[l] != s[r]) return false;
            else{
                l++; r--;
            }
        }
        return true;
    }
    bool isPalindrome(string s) {
        string str = "";
        for(int i = 0; i < s.size(); i++){
            if(s[i]-'0' >= 0 && s[i]-'0' <= 9){
                str.push_back(s[i]);
            }
            if((s[i]>= 97 && s[i] <= 122)){
                str.push_back(s[i]);
            }
            if(s[i] >= 65 && s[i] <= 90){
                str.push_back(s[i] - 'A' + 'a');
            }
        }
        
        return check(str);
    }
};